books_iterator/main.py

```diff
@@ -32,48 +37,77 @@ class Price(object):
         },
     }
 
+
     def __init__(self, amount, currency='USD'):
-        pass
+        self.amount=amount
+        self.currency=currency
 
     def __str__(self):
-        pass
+        return "Price(Decimal('{}'), '{}')".format(self.amount,self.currency)
 
     def get_currency(self):
-        pass
+        return self.currency
 
     def __add__(self, other):
         # return a new `Price` instance, representing the sum of
         # both given ones
-        pass
+        if self.currency==other.currency:
+            sum_amount=self.amount+other.amount
+        else: 
+            sum_amount=self.amount+other.get_value(self.currency)
+        return Price(sum_amount,self.currency)
 
     def __eq__(self, other):
         # compare if two prices are equal. Keep in mind that both prices
         # might have different currencies. Use the `.get_value()` function
         # to transform prices to a comparable currency.
-        pass
+        if self.currency==other.currency and self.amount==other.amount:
+            return True
+        elif self.currency!=other.currency:
+            if self.amount==other.get_value(self.currency):
+                return True
```

Comment:
this can be done in one liner:-

`return self.amount == other.get_value(self.currency)`